Engine/source/app/net/tcpObject.cpp

```diff
@@ -400,6 +405,29 @@ void TCPObject::send(const U8 *buffer, U32 len)
    Net::sendtoSocket(mTag, buffer, S32(len));
 }
 
+bool TCPObject::sendFile(const char* fileName)
+{
+	//Open the file for reading
+	FileStream readFile;
+	if(!readFile.open(fileName, Torque::FS::File::Read))
+	{
+		return false;
+	}
+
+	//Read each byte into our buffer
+	Vector<U8> buffer(readFile.getStreamSize());
+	U8 byte;
+	while(readFile.read(&byte))
+	{
+		buffer.push_back(byte);
+	}
+
+	//Send the buffer
+	send(buffer.address(), buffer.size());
```

Comment:
This was https://github.com/GarageGames/Torque3D/pull/658 - however I pulled it due to there not being "a real consensus on this about the implementation". From what I can tell the only similarity between @MusicMonkey5555 's changes (in total) and that branch are the incorporation of the bytesWritten parameter in Net::send.